File: docs/09-Greedy-Algorithms/python/activity_selection.py
def select_activities(start, end):
    # Zip and sort activities by their finish times
    activities = sorted(list(zip(start, end, range(len(start)))), key=lambda x: x[1])
    
    selected = []
    # The first activity is always selected
    selected.append(activities[0])
    last_finish_time = activities[0][1]
    
    for i in range(1, len(activities)):
        # If this activity has start time greater than or equal to the finish time of previously selected activity
        if activities[i][0] >= last_finish_time:
            selected.append(activities[i])
            last_finish_time = activities[i][1]
            
    return selected

if __name__ == "__main__":
    start = [1, 3, 0, 5, 8, 5]
    end   = [2, 4, 6, 7, 9, 9]
    
    # Expected output: (1, 2), (3, 4), (5, 7), (8, 9)
    result = select_activities(start, end)
    print("Selected activities (Start, End, Original Index):")
    for act in result:
        print(f"Activity {act[2]}: ({act[0]}, {act[1]})")

File: docs/09-Greedy-Algorithms/js/activity_selection.js
function selectActivities(activities) {
    // Sort activities by end time
    activities.sort((a, b) => a.end - b.end);

    const selected = [];
    selected.push(activities[0]);
    let lastEnd = activities[0].end;

    for (let i = 1; i < activities.length; i++) {
        if (activities[i].start >= lastEnd) {
            selected.push(activities[i]);
            lastEnd = activities[i].end;
        }
    }
    return selected;
}

// Test cases
const activities = [
    { start: 1, end: 2, id: 0 },
    { start: 3, end: 4, id: 1 },
    { start: 0, end: 6, id: 2 },
    { start: 5, end: 7, id: 3 },
    { start: 8, end: 9, id: 4 },
    { start: 5, end: 9, id: 5 }
];

const result = selectActivities(activities);
console.log("Selected activities:");
result.forEach(act => {
    console.log(`Activity ${act.id}: (${act.start}, ${act.end})`);
});
